### PR TITLE
modified server struct and added serverOptions

### PR DESCRIPTION
**Pull request by saeidalz13**
# Overview

- added `ServerOption` to encapsulate the logic of port, stage, and db.
- upgrader was taken out a global variable

### CODE REVIEW

**Comment by saeidalz13** [comment]
I found a better way compared to this.
we can put the handlers in api/handlers package. then we make the fields in `Server` struct private (lowercase fields are private).
In this case the private would not be accessible from the handlers in `Request` receiver functions.

For now we can merge this. I'll implement this in future merges

**Comment by mori-ahk** [comment]
@saeidalz13 is this PR ready for review?

**Comment by saeidalz13** [comment]
Yes it is!
There are some changes I have in mind but we'll do it after we set up the database

File: api/ws_api.go [hunk 1]
```diff
@@ -26,13 +26,27 @@ var (
...
+}
+
 type Server struct {
-	Port     *int
-	Upgrader websocket.Upgrader
-	Db       *sql.DB
-	Games    map[string]*md.Game
-	Players  map[string]*md.Player
-	mu       sync.RWMutex
+	Db      *sql.DB
```
Comment (mori-ahk) [inline api/ws_api.go:46]: Do we still need this `Db` field?

**Review by mori-ahk — APPROVED** [review]
except one comment. the rest LGTM ✅

File: api/ws_api.go (hunk 1 above)
Comment (saeidalz13) [reply api/ws_api.go:46]: Yea we don't need that. after we want to set up the database, we actually separate the handlers and DB doesn't have to be in server struct

**Pull request merged**